ci: disabling most of the hardfork-specific tests

evertonfraga opened the pull request:
As promised, this PR disables most of the hardfork specific tests, to favor job concurrency. 

It can be easily reverted, by un-commenting those lines.

evertonfraga commented:
@jochem-brouwer double-checking: what are the most important HFs to test here? We can keep to 2 or 3 for brevity.

@holgerd77 After choosing the HFs, can I tweak the required status checks for `master` branch?

<img width="865" alt="image" src="https://user-images.githubusercontent.com/47108/98958719-91a15900-24d0-11eb-8f42-b7002aedccf6.png">

jochem-brouwer commented:
I'd say the most important are `MuirGlacier` (mainnet), `Istanbul` (latest hard fork w/h difficulty EIP -> `MuirGlacier`) since most networks run on one of both. All other forks are "outdated" so they don't need to be stress tested on every PR (except if changes are specific for that HF).

jochem-brouwer commented on .github/workflows/vm-pr.yml:
```diff
@@ -53,15 +53,15 @@ jobs:
         fork: [
           'Berlin',
-          'MuirGlacier',
+          # 'MuirGlacier',
```
I'd say we also include `MuirGlacier` since this is what currently runs on mainnet.

evertonfraga replied:
Thanks! the changes are in place.

jochem-brouwer reviewed (approved):
Yup, great!

evertonfraga commented:
I have updated the required checks according to the state and blockchain I have just changed.

Closing and reopening PR to try to see the Required Checks change in effect.

holgerd77 commented:
>All other forks are "outdated" so they don't need to be stress tested on every PR (except if changes are specific for that HF).

I would be careful with this, there are plenty of ways to break old HF code with lastest-fork-feature-adding PRs, we just had an example for this with the broken `Constantinople` tests fixed in #931. Should be ok though I would say with #951, we should nevertheless be aware of this and in doubt rather trigger the test runs for all the hardforks.
